lib/cli.js

```diff
@@ -182,6 +183,15 @@ const cli = {
 
             log.info(JSON.stringify(fileConfig, null, "  "));
             return 0;
+        } else if (currentOptions.info) {
+            log.info("\n Environment Information:-");
+            envinfo
+                .run({
+                    System: ["OS", "CPU"],
+                    Binaries: ["Node", "Yarn", "npm"],
+                    npmGlobalPackages: ["eslint"]
+                })
+                .then(log.info);
```

Comment:
Does `envinfo` have a synchronous API option? It looks like we'd have a dangling promise when it's called [here](https://github.com/eslint/eslint/blob/master/bin/eslint.js#L81).

Reply:
Fetching that system config information is meant to take time while something else is intended to be performed. Hence, they don't offer sync versions as of now.

Reply:
I don't think it's a good idea to have a dangling, unhandled promise here, since it can lead to a lot of confusion/hard to debug behavior (as demonstrated by my comment about the test below). If someone were to look at the code, it wouldn't immediately be clear that the function is doing something asynchronous.

We've talked about making other parts of ESLint asynchronous (See [RFC](https://github.com/eslint/rfcs/pull/4), and I think that could lead to some potential race conditions with the logging logic.